tests/phpunit/tests/dependencies/scripts.php

```diff
@@ -276,6 +280,128 @@ public function test_delayed_dependent_with_blocking_dependency_not_enqueued( $s
+					wp_enqueue_script( 'foo', 'https://example.com/foo.js', array(), null, array( 'strategy' => 'defer' ) );
+					wp_enqueue_script( 'bar', 'https://example.com/bar.js', array( 'bar' ), null, array( 'strategy' => 'defer' ) );
+					return 'foo';
+				},
+				'async_only' => false,
+				'expected'   => true,
+			),
+			'one_blocking_dependent'               => array(
+				'set_up'     => static function () {
+					wp_enqueue_script( 'foo', 'https://example.com/foo.js', array(), null, array( 'strategy' => 'defer' ) );
+					wp_enqueue_script( 'bar', 'https://example.com/bar.js', array( 'foo' ), null );
+					return 'foo';
+				},
+				'async_only' => false,
+				'expected'   => false,
+			),
+			'one_blocking_dependent_not_enqueued'  => array(
+				'set_up'     => static function () {
+					wp_enqueue_script( 'foo', 'https://example.com/foo.js', array(), null, array( 'strategy' => 'defer' ) );
+					wp_register_script( 'bar', 'https://example.com/bar.js', array( 'foo' ), null );
+					return 'foo';
+				},
+				'async_only' => false,
+				'expected'   => true, // Because bar was not enqueued, only foo was.
+			),
+			'two_delayed_dependents'               => array(
+				'set_up'     => static function () {
+					wp_enqueue_script( 'foo', 'https://example.com/foo.js', array(), null, array( 'strategy' => 'defer' ) );
+					wp_enqueue_script( 'bar', 'https://example.com/bar.js', array( 'foo' ), null, array( 'strategy' => 'defer' ) );
+					wp_enqueue_script( 'baz', 'https://example.com/baz.js', array( 'foo' ), null, array( 'strategy' => 'defer' ) );
+					return 'foo';
+				},
+				'async_only' => false,
+				'expected'   => true,
+			),
+			'recursion_not_delayed'                => array(
+				'set_up'     => static function () {
+					wp_enqueue_script( 'foo', 'https://example.com/foo.js', array( 'foo' ), null );
+					return 'foo';
+				},
+				'async_only' => false,
+				'expected'   => false,
+			),
+			'recursion_yes_delayed'                => array(
+				'set_up'     => static function () {
+					wp_enqueue_script( 'foo', 'https://example.com/foo.js', array( 'foo' ), null, array( 'strategy' => 'defer' ) );
+					return 'foo';
+				},
+				'async_only' => false,
+				'expected'   => true,
+			),
+			'recursion_triple_level'               => array(
+				'set_up'     => static function () {
+					wp_enqueue_script( 'foo', 'https://example.com/foo.js', array( 'baz' ), null, array( 'strategy' => 'defer' ) );
+					wp_enqueue_script( 'bar', 'https://example.com/bar.js', array( 'foo' ), null, array( 'strategy' => 'defer' ) );
+					wp_enqueue_script( 'baz', 'https://example.com/bar.js', array( 'bar' ), null, array( 'strategy' => 'defer' ) );
+					return 'foo';
+				},
+				'async_only' => false,
+				'expected'   => true,
+			),
+			'async_only_with_defer_dependency'     => array(
+				'set_up'     => static function () {
+					wp_enqueue_script( 'foo', 'https://example.com/foo.js', array( 'baz' ), null, array( 'strategy' => 'async' ) );
```

Comment:
Is the `baz` dependency supposed to be in this test?
```suggestion
					wp_enqueue_script( 'foo', 'https://example.com/foo.js', array(), null, array( 'strategy' => 'async' ) );
```